### ISSUE TITLE
Error detected while processing function <SNR>212_RefileAndClose[5]..dotoo#move_headline

### ISSUE THREAD

**Issue by unknownbreaker** [dhruvasagar/vim-dotoo#62]
I get an error when attempting to close out of the Notes template.

1. I setup a dummy dotoo file `dotootest.dotoo` and then hit `gC`.
2. Typed in random note and did `:w`. Saves OK.
3. When doing `:q`, I get an error message.

![image](https://user-images.githubusercontent.com/6026454/74844856-183b5800-52fc-11ea-95b9-fe16658969fd.png)

**Comment by Gavinok** [comment]
What does the template look like?

**Comment by dhruvasagar** [comment]
Are you sure you are on the latest ? I think I had pushed something that broke but had fixed it later. Can you confirm if you're facing this issue on the latest commit ? @unknownbreaker Thanks

**Comment by dhruvasagar** [comment]
Are you sure you are on the latest ? I think I had pushed something that broke but had fixed it later. Can you confirm if you're facing this issue on the latest commit ? @unknownbreaker Thanks

**Comment by dhruvasagar** [comment]
Also can you confirm if the path `~/Documents/dotoo-files` exists ? If not you should create that, or you can override the configuration `g:dotoo#agenda#files`, kindly refer to the inline vim help for more information

**Comment by cstsunfu** [comment]
I have this issue too. It not always occurs in my vim.
My configure is:
 ```     
        let g:dotoo#agenda#files = ['/Users/sun/Documents/dotoo-files/*.dotoo']
        let g:dotoo#capture#refile = expand( '/Users/sun/Documents/dotoo-files/refile.dotoo')
        let g:dotoo#capture#templates = {
              \ 't': {
              \   'description': 'Todo',
              \   'lines': [
              \     '* TODO %?',
              \     'DEADLINE: [%(strftime(g:dotoo#time#datetime_format))]'
              \   ],
              \  'target': 'refile:Tasks'
              \ },
              \ 'n': {
              \   'description': 'Note',
              \   'lines': ['* %? :NOTE:'],
              \ },
              \ 'm': {
              \   'description': 'Meeting',
              \   'lines': ['* MEETING with %? :MEETING:'],
              \ },
              \ 'p': {
              \   'description': 'Phone call',
              \   'lines': ['* PHONE %? :PHONE:'],
              \ },
              \ 'h': {
              \   'description': 'Habit',
              \   'lines': [
              \     '* NEXT %?',
              \     'SCHEDULED: [%(strftime(g:dotoo#time#date_day_format)) +1m]',
              \     ':PROPERTIES:',
              \     ':STYLE: habit',
              \     ':REPEAT_TO_STATE: NEXT',
              \     ':END:'
              \   ]
              \ }
              \}
```
I think this issue maybe related to the temporary file.  When I capture one todo item, this plugin will open one temporary file like "/private/var/folders/kw/_rs242hj7h5d2lx5344tww4c0000gp/T/nvim5N3yPI", but this file is readonly.

OS X & neovim v0.4.3

**Comment by dhruvasagar** [comment]
@cstsunfu Would you be able to answer my questions I stated above ?

**Comment by cstsunfu** [comment]
Yes, the path of g:dotoo#agenda#files exist on my system, and my plugin version is the newest.
This issue not always occurs in my vim. In the same environment, sometimes it works well, sometimes it's wrong, so I didn't know what triggers this issue.

PS: The temporary file is writable when I use :w to save, but when I use :q to quit the capture, this file will be readonly.

**Comment by dhruvasagar** [comment]
@cstsunfu I believe this should not be an issue with on the latest version of vim-dotoo. Kindly test again and let me know.
